packages/notifier/src/publish-kit.js

```diff
@@ -316,20 +316,21 @@ const provideDurablePublishKitEphemeralData = (state, facets) => {
  * Extends the sequence of results.
  *
  * @param {{state: DurablePublishKitState, facets: PublishKit<*>}} context
- * @param {boolean} done
  * @param {any} value
- * @param {any} [rejection]
+ * @param {DurablePublishKitState["status"]} [targetStatus]
  */
-const advanceDurablePublishKit = (context, done, value, rejection) => {
+const advanceDurablePublishKit = (context, value, targetStatus = 'live') => {
   const { state, facets } = context;
   const { valueDurability, status } = state;
   if (status !== 'live') {
     throw new Error('Cannot update state after termination.');
   }
-  if (done || valueDurability === 'mandatory') {
+  const done = targetStatus !== 'live';
+  if (targetStatus === 'failed') {
+    canBeDurable(value) ||
+      Fail`Cannot accept non-durable fail reason: ${value}`;
+  } else if (done || valueDurability === 'mandatory') {
     canBeDurable(value) || Fail`Cannot accept non-durable value: ${value}`;
```

Comment:
I'm not sure it's worth keeping a distinct message for `fail(nonDurableReason)` vs. `finish(nonDurableFinalValue)` and `publish(nonDurableValue)`. If not, we could simplify this.
```suggestion
  if (done || valueDurability === 'mandatory') {
    canBeDurable(value) || Fail`Cannot accept non-durable value: ${value}`;
```

Reply:
The stack trace should show which case it was. The only value I can think of is for a test to be able to verify which failure, but it could do that by exercising both paths. So +1 to the simplification.